olp-cpp-sdk-core/include/olp/core/client/OlpClientSettings.h

```diff
@@ -248,6 +249,20 @@ struct CORE_API ApiLookupSettings {
    */
   using LookupEndpointProvider = std::function<std::string(const std::string&)>;
 
+  /**
+   * @brief The type alias of the catalog endpoint provider function.
+   *
+   * Catalogs which already have an static URL or can be accessed through a
+   * proxy service can input the URL provider here and it will be taken by the
+   * ApiLookupClient and returned directly to the caller without any request to
+   * the Lookup API service.
+   *
+   * @note Return empty string in case of an invalid or unknown catalog.
+   * @note This call should be synchronous without any tasks scheduled on the
+   * TaskScheduler as this might result in a dead-lock.
+   */
```

Comment:
```suggestion
  /**
   * @brief The type alias of the catalog endpoint provider function.
   *
   * Catalogs that have a static URL or can be accessed through
   * a proxy service can input the URL provider here. This URL provider is taken by
   * the `ApiLookupClient` and returned directly to the caller without any requests to
   * the API Lookup Service.
   *
   * @note This call should be synchronous without any tasks scheduled in
   * `TaskScheduler` as it might result in a dead-lock.
   *
   * @return An empty string if the catalog is invalid or unknown.
   */
```